cpp/src/reduction.cpp

```diff
@@ -81,22 +98,67 @@ class ReduceLocalTask : public Task<ReduceLocalTask, OpCode::ReduceLocal> {
 
   static constexpr auto GPU_VARIANT_OPTIONS = legate::VariantOptions{}.with_has_allocations(true);
 
+  static void cpu_variant(legate::TaskContext context)
+  {
+    TaskContext ctx{context};
+
+    const auto input = argument::get_next_input<PhysicalColumn>(ctx);
+    auto op          = argument::get_next_scalar<std::string>(ctx);
+    auto finalize    = argument::get_next_scalar<bool>(ctx);
+    auto initial     = argument::get_next_scalar<bool>(ctx);
+    auto output      = argument::get_next_output<PhysicalColumn>(ctx);
+
+    auto array = input.arrow_array_view();
+    std::shared_ptr<arrow::Array> result_array;
+    if (op == "count_valid") {
+      assert(!initial);
+      if (!finalize) {
+        auto count   = std::make_shared<arrow::Int64Scalar>(array->length() - array->null_count());
+        result_array = ARROW_RESULT(arrow::MakeArrayFromScalar(*count, 1));
+      } else {
+        auto sum     = ARROW_RESULT(arrow::compute::Sum(array));
+        result_array = ARROW_RESULT(arrow::MakeArrayFromScalar(*sum.scalar(), 1));
+      }
+    } else {
+      if (initial) {
+        auto initial_col     = argument::get_next_input<PhysicalColumn>(ctx);
+        auto initial_array   = initial_col.arrow_array_view();
+        auto result          = ARROW_RESULT(arrow::compute::CallFunction(op, {array})).scalar();
+        auto result_as_array = ARROW_RESULT(arrow::MakeArrayFromScalar(*result, 1));
+        // Combine two arrays and reduce again
```

Comment:
That's too bad, but if an "initial" isn't explicitly supported, I guess this is the pragmatic thing.  Special casing it for specific operation doesn't seem useful...